horovod/common/gloo_context.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2019 Uber Technologies, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+
+#ifndef HOROVOD_GLOO_CONTEXT_H
+#define HOROVOD_GLOO_CONTEXT_H
+
+#include "gloo/context.h"
+#include "mpi.h"
+
+#include "common.h"
+
+namespace horovod {
+namespace common {
+
+struct GlooContext {
+  void InitializeFromMPI(const MPI_Comm &mpi_comm, const char* gloo_iface);
+
+  void Finalize();
+
+  void Initialize(const MPI_Comm &mpi_comm, bool gloo_data, bool
+  gloo_control, const char* gloo_iface);
+
+  std::shared_ptr<gloo::Context> ctx;
+
+  bool data_transfer_enabled;
+
+  bool control_transfer_enabled;
```

Comment:
Is this actually used at this point, or just for future?

Reply:
These will be used in the future to distinguish data transferring and control info transferring.